WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift

```diff
@@ -112,17 +119,47 @@ final class BlazeConfirmPaymentViewModel: ObservableObject {
         }
 
         analytics.track(event: .Blaze.Payment.submitCampaignTapped())
-        shouldDisplayCampaignCreationError = false
+        campaignCreationError = nil
         isCreatingCampaign = true
         do {
-            let updatedDetails = campaignInfo.copy(paymentMethodID: selectedPaymentMethod.id)
-            try await requestCampaignCreation(details: updatedDetails)
+            // Prepare image for campaign
+            let campaignMedia: Media
+            switch image.source {
+            case .asset(let asset):
+                do {
+                    campaignMedia = try await uploadPendingImage(asset)
+                } catch {
+                    DDLogError("⛔️ Error uploading campaign image: \(error)")
+                    throw BlazeCampaignCreationError.failedToUploadCampaignImage
+                }
+            case .media(let media):
+                campaignMedia = media
+            case .productImage(let image):
+                do {
+                    campaignMedia = try await retrieveMedia(mediaID: image.imageID)
+                } catch {
+                    DDLogError("⛔️ Error fetching product image's Media: \(error)")
+                    throw BlazeCampaignCreationError.failedToFetchCampaignImage
+                }
+            }
+
+            var updatedDetails = campaignInfo
+            // Set image URL and mimeType
+            updatedDetails = updatedDetails.copy(mainImage: .init(url: campaignMedia.src, mimeType: campaignMedia.mimeType))
+            // Set payment method ID
+            updatedDetails = updatedDetails.copy(paymentMethodID: selectedPaymentMethod.id)
```

Comment:
Suggestion: this can be simplify to one call:

```suggestion
            // Set payment method ID, image URL and mimeType
            let updatedDetails = campaignInfo
                .copy(paymentMethodID: selectedPaymentMethod.id,
                      mainImage: .init(url: campaignMedia.src, mimeType: campaignMedia.mimeType))
```

Reply:
Nice. Done in https://github.com/woocommerce/woocommerce-ios/pull/12006/commits/2c0dfb062100be3cd8f0897f133441b2ee3f68f4